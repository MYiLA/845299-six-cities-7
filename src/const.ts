export enum AppRoute {
  MAIN = 'main',
  LOGIN = 'login',
  FAVORITES = 'favorites',
  OFFER = 'offer',
  DEFAULT_CITY = 'Paris'
};

export enum APIRoute {
  HOTELS = '/hotels',
  FAVORITE = '/favorite',
  COMMENTS = '/comments',
  LOGIN = '/login',
  LOGOUT = '/logout',
};

export enum CardType {
  FAVORITES = 'favorites',
  CITIES = 'cities',
  NEAR_PLACES = 'near-places'
}

export enum AuthorizationStatus {
  AUTH = 'AUTH',
  NO_AUTH = 'NO_AUTH',
  UNKNOWN = 'UNKNOWN',
};

export const Cities = [
  {
    location: {
      latitude: 48.85,
      longitude: 2.35,
      zoom: 10,
    },
    name: 'Paris',
  },
  {
    location: {
      latitude: 50.9375,
      longitude: 6.9603,
      zoom: 10,
    },
    name: 'Cologne',
  },
  {
    location: {
      latitude: 50.8503,
      longitude: 4.3517,
      zoom: 10,
    },
    name: 'Brussels',
  },
  {
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10,
    },
    name: 'Amsterdam',
  },
  {
    location: {
      latitude: 53.5511,
      longitude: 9.9937,
      zoom: 10,
    },
    name: 'Hamburg',
  },
  {
    location: {
      latitude: 51.2277,
      longitude: 6.7735,
      zoom: 10,
    },
    name: 'Dusseldorf',
  },
];

export const RegularExpression = {
  EMAIL: /^[^\s@]+@[^\s@]+\.[^\s@]+$/,
  ONE_SIMBOL: /\S/,
}
