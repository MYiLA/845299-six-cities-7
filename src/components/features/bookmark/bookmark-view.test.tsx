import { fireEvent, render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { BookmarkType, SNAP_DESC } from '../../../const';
import BookmarkView from './bookmark-view';

let history = {} as MemoryHistory<unknown>;
let onSetFavorite = () => null;
const isFavorite = true;
const type = BookmarkType.PROPERTY;
const width = 31;
const height = 33;

describe('поведение компонента bookmark-view', () => {
  beforeEach(() => {
    history = createMemoryHistory();
    onSetFavorite = jest.fn();
  });

  it('делает разметку на экране', () => {
    const { container } = render(
      <Router history={history}>
        <BookmarkView
          isFavorite={isFavorite}
          onSetFavorite={onSetFavorite}
          type={type}
          width={width}
          height={height}
        />
      </Router>,
    );

    expect(container.querySelector(`.${type}__bookmark-button--active`)).toBeTruthy();
    const bookmarkIcon = container.querySelector('.property__bookmark-icon');
    expect(bookmarkIcon).toBeTruthy();
    if (bookmarkIcon !== null) {
      expect(bookmarkIcon.getAttribute('width')).toBe('31');
      expect(bookmarkIcon.getAttribute('height')).toBe('33');
    }
  });

  it('обрабатывает клик по флажку-кнопке', () => {
    const { container } = render(
      <Router history={history}>
        <BookmarkView
          isFavorite={isFavorite}
          onSetFavorite={onSetFavorite}
          type={type}
          width={width}
          height={height}
        />
      </Router>,
    );

    const button = container.querySelector('.button');
    if (button !== null) {
      fireEvent.click(button);
    }
    expect(onSetFavorite).toBeCalled();
  });

  it(SNAP_DESC, () => {
    const { container } = render(
      <BookmarkView
        isFavorite={isFavorite}
        onSetFavorite={onSetFavorite}
        type={type}
        width={width}
        height={height}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
