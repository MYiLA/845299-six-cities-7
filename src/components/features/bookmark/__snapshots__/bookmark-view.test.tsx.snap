// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`поведение компонента bookmark-view нет изменений в разметке, по сравнению с прошлым тестом 1`] = `
<div>
  <button
    class="property__bookmark-button button property__bookmark-button--active"
    type="button"
  >
    <svg
      class="property__bookmark-icon"
      height="33"
      width="31"
    >
      <use
        xlink:href="#icon-bookmark"
      />
    </svg>
    <span
      class="visually-hidden"
    >
      To bookmarks
    </span>
  </button>
</div>
`;
